File: backend/app/schemas/batch.py
"""Schemas Pydantic para o endpoint de upload de CSV em lote."""

from pydantic import BaseModel


class BatchRowResult(BaseModel):
    row_index: int
    predicted_class: str
    probability_attack: float
    risk_level: str


class BatchRowError(BaseModel):
    row_index: int
    error: str


class BatchPredictionResponse(BaseModel):
    total_rows: int
    processed_rows: int
    failed_rows: int
    total_attacks: int
    total_normal: int
    attack_rate: float
    processing_time_ms: float
    results: list[BatchRowResult]
    errors: list[BatchRowError]

File: frontend/src/types/batch.ts
export interface BatchRowResult {
  row_index: number;
  predicted_class: string;
  probability_attack: number;
  risk_level: string;
}

export interface BatchRowError {
  row_index: number;
  error: string;
}

export interface BatchPredictionResponse {
  total_rows: number;
  processed_rows: number;
  failed_rows: number;
  total_attacks: number;
  total_normal: number;
  attack_rate: number;
  processing_time_ms: number;
  results: BatchRowResult[];
  errors: BatchRowError[];
}